packages/server/utils/analytics/amplitude/AmplitudeAnalytics.ts

```diff
@@ -0,0 +1,37 @@
+import {init, track} from '@amplitude/analytics-node'
+import {AnalyticsEvent} from '../analytics'
+import PROD from '../../../PROD'
+import NullableDataLoader from '../../../dataloader/NullableDataLoader'
+import IUser from '../../../postgres/types/IUser'
+
+const {AMPLITUDE_WRITE_KEY} = process.env
+
+/**
+ * Wrapper for amplitude providing a more typesafe interface
+ */
+export class AmplitudeAnalytics {
+  constructor() {
+    // used as a failsafe for PPMIs
+    if (!AMPLITUDE_WRITE_KEY) return
+    init(AMPLITUDE_WRITE_KEY, {
+      flushQueueSize: PROD ? 20 : 1,
+      optOut: !AMPLITUDE_WRITE_KEY
+    })
+  }
+
+  async track(
+    userId: string,
+    event: AnalyticsEvent,
+    dataloader: NullableDataLoader<string, IUser, string>,
```

Comment:
+1 Usually we would pass `dataLoader: DataLoaderWorker` here.

Reply:
I got a type error when trying to use DataLoaderWorker as it had a type of `CacheWorker<RootDataLoader>` but `getDataLoader()` returns a type of `CacheWorker<DataLoaderBase>`.

I used `CacheWorker<DataLoaderBase>` for the track() interface to get this merged quickly, but can revisit later if necessary.